### PR TITLE
Full Yellow Local - Fix GCP tests in 4.2

### LINKED ISSUE #1532: Full Yellow Local - Fix GCloud tests in 4.2
## Description

This issue found while working on [Research: Check the integration tests status in 4.2.0-RC8](https://github.com/wazuh/wazuh-qa/issues/1493). We detect that GCloud Tests aren't working successfully.
It's required to investigate the root cause of each Failed. 

### Test Execution - Results

| Manager  |  **Local** | **Jenkins**  |  **Status**|
|:--:|:--: |:--:|:--:|
|**Test Execution 1**| [Results](https://github.com/wazuh/wazuh-qa/files/6750323/Test_gcloud_r1.log)| https://devel.ci.wazuh.info/view/Tests/job/Test_integration/876/  |    🔴  | 
|**Test Execution 2**| [Results](https://github.com/wazuh/wazuh-qa/files/6750379/Test_gcloud_r2.log)| https://devel.ci.wazuh.info/view/Tests/job/Test_integration/918/  |    🔴  | 
|**Test Execution 3**| [Results](https://github.com/wazuh/wazuh-qa/files/6750395/Test_gcloud_r3.log) | https://devel.ci.wazuh.info/view/Tests/job/Test_integration/950/ |    🔴  | 

### Package Details
 
|  Jenkins branch  | QA branch   |  Version 	|   Revision | 
|- |- |- |- |
|  4.2 |   1516-4.2.0-full-green | 4.2  |  1.1493 | 

Type | Format | Architecture | Version | Revision | Tag | File name
-- | -- | -- | -- | -- | -- | --
Server | rpm | x86_64 | 4.2.0 | 40207 | v4.2.0-rc8 | wazuh-manager-4.2.0-1.1493.x86_64.rpm

### Environment 

Provider| Box| OS| CPU| Memory | 
--|--|--|--|--|
Vagrant | centos/8 | CentOS 8 | 2 | 1024 |

### Setup Local_internal_opcion.conf
```
analysisd.debug=2
wazuh_modules.debug=2
monitord.rotate_log=0  
```

## Development 

In order to finish this issue the following tasks should be fulfilled:

- [x] Fix `GCloud` integration tests.
  - [x] Fix `test_configuration/test_invalid` for cases:
       - [x] `get_configuration15`
       - [x] `get_configuration16`
       - [x] `get_configuration19`       
       - [x] `get_configuration20`
       - [x] `get_configuration21`             
       - [x] `get_configuration22`
       - [x] `get_configuration23`
       - [x] `get_configuration24`       
       - [x] `get_configuration25`
       
   <hr>

  - [x] Fix `test_functioning/test_day_wday` for cases:
       - [x] `get_configuration0-tags_to_apply0`
       - [x] `get_configuration1-tags_to_apply1`
       - [x] `get_configuration2-tags_to_apply2`

|  Type | Description  |
|--|--|
|Test Path | test_gcloud/test_functioning/test_day_wday.py
| Consistent | Yes 
| Test Execution | 3/3 executions failed
| Cases |   get_configuration0-tags_to_apply0, get_configuration1-tags_to_apply1, get_configuration2-tags_to_apply2
| Cases Fails | 3
| Summary |  Did not receive expected "Starting fetching of logs" event
| Error message |  Timeout
```
        wazuh_log_monitor.start(timeout=global_parameters.default_timeout + 120,
                                callback=callback_detect_start_fetching_logs,
                                accum_results=1,
                               error_message='Did not receive expected '
                                              '"Starting fetching of logs" event').result()
```
<hr>

  - [x] Fix `test_functioning/test_max_messages` for cases:
       - [x] `get_configuration0-30`
       - [x] `get_configuration0-100`
       - [x] `get_configuration0-120`

|  Type | Description  |
|--|--|
|Test Path | test_gcloud/test_functioning/test_max_messages.py
| Consistent | Yes 
| Test Execution | 3/3 executions failed
| Cases |   get_configuration0-30, get_configuration0-100, get_configuration0-120
| Cases Fails | 3
| Summary |  Did not receive expected - INFO - Received and acknowledged x messages
| Error message |  Timeout

```
        if nmessages <= max_messages:
            number_pulled = wazuh_log_monitor.start(timeout=global_parameters.default_timeout + time_interval + 5,
                                                    callback=callback_received_messages_number,
                                                    accum_results=1,
                                                  error_message='Did not receive expected - INFO - Received and acknowledged x messages').result()
```

<hr>

  - [x] Fix `test_functioning/test_rules` for cases:
       - [x] `get_configuration0`

|  Type | Description  |
|--|--|
|Test Path | test_gcloud/test_functioning/test_rules.py
| Consistent | Yes 
| Test Execution | 3/3 executions failed
| Cases |   get_configuration0
| Cases Fails | 1
| Summary |  "Resource not found (resource=wazuh-pubsub).","grpc_status":5}"
| Error message |  google.api_core.exceptions.NotFound: 404 Resource not found (resource=wazuh-pubsub).
```
	status = StatusCode.NOT_FOUND
	details = "Resource not found (resource=wazuh-pubsub)."
	debug_error_string = "{"created":"@1626085199.275600532","description":"Error received from peer ipv4:142.250.201.74:443","file":"src/core/lib/surface/call.cc","file_line":1067,"grpc_message":"Resource not found (resource=wazuh-pubsub).","grpc_status":5}"
```

- [x] Full Yellow in `GCloud` tests for Manager.
- [x] Documentation of any important change done for these tests or the used tools.
- [ ] Merge with `1516-4.2.0-full-green`.

### PR DESCRIPTION
**Pull request by Molter73**
|Related issue|
|---|
|#1532|

## Description

Closes #1532

This PR fixes some of the GCP integrations tests that were failing in order to achieve a full green/yellow locally in a reliable way.

Some minor improvements to the code have also been made, namely:
- Added the possibility of using a configuration file when running tests instead of using CLI options.
- Created a synchronous variant of the `publish` function found in `gcloud.py`
- Created a fixture for publishing messages to GCP. 
- Moved repeated validation of global parameters to a fixture.

## Configuration options

All tests were run on a VM with the following characteristics:
- Centos 8 (vagrant box: centos/8)
- 2 cores
- 2 GB of RAM
- Wazuh Manager v4.2.0-rc8

## Testing
- [R1](https://github.com/wazuh/wazuh-qa/files/6884817/gcloud-0.zip) 🟢 
- [R2](https://github.com/wazuh/wazuh-qa/files/6884816/gcloud-1.zip) 🟢 
- [R3](https://github.com/wazuh/wazuh-qa/files/6884815/gcloud-2.zip) 🟢

### CODE REVIEW

File: tests/integration/test_gcloud/test_functionality/test_day_wday.py [hunk 1]
```diff
@@ -97,53 +78,31 @@ def test_day_wday(tags_to_apply, get_configuration, configure_environment,
     These tests verify the module starts to pull according to the day of the week
     or month and time.
     """
-    check_apply_test(tags_to_apply, get_configuration['tags'])
+    def get_next_scan(next_scan_time: str):
+        next_scan_time = next_scan_time_log.split(" ")
```
Comment (Rebits) [inline tests/integration/test_gcloud/test_functionality/test_day_wday.py:82]:
```suggestion
        next_scan_time = next_scan_time_log.split()
```

File: tests/integration/test_gcloud/test_functionality/test_day_wday.py [hunk 2]
```diff
@@ -97,53 +78,31 @@ def test_day_wday(tags_to_apply, get_configuration, configure_environment,
...
+        date = next_scan_time[0].split("/")
+        hour = next_scan_time[1].split(":")
```
Comment (Rebits) [inline tests/integration/test_gcloud/test_functionality/test_day_wday.py:84]:
```suggestion
        date = next_scan_time[0].split('/')
        hour = next_scan_time[1].split(':')
```

**Review by Rebits — COMMENTED** [review]
Awesome job, minor changes are required.

File: tests/integration/test_gcloud/test_functionality/test_day_wday.py (hunk 1 above)
Comment (Molter73) [reply tests/integration/test_gcloud/test_functionality/test_day_wday.py]: Done in commit [b855b5c](https://github.com/wazuh/wazuh-qa/pull/1621/commits/b855b5c20954769f6d12e6fbffa8bdcb7b22bf90)

File: tests/integration/test_gcloud/test_functionality/test_day_wday.py (hunk 2 above)
Comment (Molter73) [reply tests/integration/test_gcloud/test_functionality/test_day_wday.py]: Done in commit [b855b5c](https://github.com/wazuh/wazuh-qa/pull/1621/commits/b855b5c20954769f6d12e6fbffa8bdcb7b22bf90)

**Comment by Rebits** [comment]
### Update 23/07/2021

After some testing we have a check that these tests fail for CentOS8:

[R1](https://github.com/wazuh/wazuh-qa/files/6868138/report.zip)

**Review by Rebits — APPROVED** [review]
LGTM

**Comment by Rebits** [comment]
### Update 23/07/2021
Using the following `local_internal_options`, these tests work perfectly.
```
wazuh_modules.debug=2
analysisd.debug=2
monitord.rotate_log=0
```

- [R1](https://github.com/wazuh/wazuh-qa/files/6869463/r1_gcloud.zip) :green_circle: 
- [R2](https://github.com/wazuh/wazuh-qa/files/6869464/r2_gcloud.zip) :green_circle: 
- [R3](https://github.com/wazuh/wazuh-qa/files/6869465/r3_gcloud.zip) :green_circle:

**Review by snaow — APPROVED** [review]

**Comment by snaow** [comment]
LGTM

||Status|Reports|
|:--:|:--:|:--:|
|R1| 🟢   | [report_gcloud_r2.zip](https://github.com/wazuh/wazuh-qa/files/7033750/report_gcloud_r2.zip) |

**Pull request merged**